ibis/expr/types/arrays.py

```diff
@@ -1131,4 +1131,6 @@ def array(values: Iterable[V]) -> ArrayValue:
     │ [3, 47]              │
     └──────────────────────┘
     """
+    if isinstance(values, ir.ArrayValue):
+        return values
```

Comment:
This functionality seems unrelated to the original issue and needs tests. Can you add this feature in a follow up PR?